packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts

```diff
@@ -65,7 +65,7 @@ export interface LoadBalancedServiceBaseProps {
    *
    * @default - No environment variables.
    */
-  readonly environment?: { [key: string]: string };
+  readonly environment?: { [key: string]: ecs.EnvironmentValue };
```

Comment:
I feel like we might want to use some token magic here.

FWIW, we should have Tokenized representations of secret values already. Can we not use those in some way?

Reply:
For the `Secrets` property, CF expects the ARN of the secret/ssm param, the container will pull that value at startup and set it as env var avoiding passing secrets in clear text.

The class `EnvironmentValue` will feed either `environment` or `secrets` based on the type. This gives a better user experience (higher level of abstraction) than having to specify `environment` and/or `secrets` manually (both will become env vars at the end).

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-environment
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-secret.html

The support for the `Secrets` property was added in the latest release of CF (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html).